spec/visual/samples/epp

```diff
@@ -0,0 +1,24 @@
+<%- | Boolean $keys_enable,
+      String  $keys_file,
+      Array   $keys_trusted,
+      String  $keys_requestkey,
+      String  $keys_controlkey
+| -%>
+<%# Parameter tag ↑ -%>
+
+<%# Non-printing tag ↓ -%>
+<% if $keys_enable { -%>
+
+<%# Expression-printing tag ↓ -%>
+keys <%= $keys_file %>
+<% unless $keys_trusted =~ Array[Data,0,0] { -%>
+trustedkey <%= $keys_trusted.join(' ') %>
+<% } -%>
+<% if $keys_requestkey =~ String[1] { -%>
+requestkey <%= $keys_requestkey %>
+<% } -%>
+<% if $keys_controlkey =~ String[1] { -%>
+controlkey <%= $keys_controlkey %>
+<% } -%>
+
+<% } -%>
```

Comment:
Could you include some HTML here as a sanity check that the delegation to parent is indeed working?

Reply:
I'm not sure if I'm completely following here? EPP delegates to Puppet. (And shows the fact in the visual sample by using puppet variables and typing)

Reply:
This is a subclass of TemplateLexer. That lexer has two objects to which it delegates: `parent` and the lexer for the 'underlying' language. So for ERB, the parent language is HTML (which is the default unless specified otherwise) and the underlying language is Ruby.

Is EPP not intended to work that way? Your final rule in the `:root` state delegates to `parent`. Should it not be doing that?

Reply:
The demo file uses HTML-style `<title>` tags so I presumed it was intentional to be delegating to the default parent lexer.

Reply:
Well, you *could* theoretically use EPP for templating HTML, though I don't - off the top of my head - know anything that actually does so. EPP is mostly used for templating configuration files through Puppet, so INI, YAML, shell vars, that sort of thing.

The demo file was basically a direct copy paste of something else that I switched to EPP semantics, but kept the template itself, so perhaps I should rewrite that one to be more reasonable to what EPP is more likely to be used for.

Reply:
I hope you don't mind but I've force pushed some changes to your branch. In sum:

1. Given how slow we were to get to this, the underlying Rouge code on which it's based is old. I rebased against the current master. You should now be able to run the tests just using `rake` (you may need to get rid of your Gemfile.lock file and reinstall the gems).

2. I made the PlainText lexer the default parent. For the generic template lexer, it makes sense for this to be HTML but based on what you were saying and my own cursory examination of the Puppet documentation, there's no obvious language to make the default. In such a situation, plaintext seems the best choice to me.

While that puts this in good shape I did come across a problem when loading up the visual test app (`bundle exec rackup`) and examining the visual sample. There's an error in Puppet! It doesn't match dots between objects and methods! I'll have a look into that and see if I can't submit a PR to fix it. Forgive me for the delay!